wrangler-core/src/main/java/io/cdap/directives/column/ExplodeRecord.java

```diff
@@ -0,0 +1,141 @@
+import io.cdap.wrangler.api.parser.Text;
+import io.cdap.wrangler.api.parser.TokenType;
+import io.cdap.wrangler.api.parser.UsageDefinition;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A directive for splitting a RowType ("Record") Column into multiple Columns.
+ *
+ * The directive takes multiple columns of type "Row" as brings all its elements up a level.
+ * Topmost elements will be set under new columns prefixed with the name of the parent column
+ * followed by an underscore. Existing columns with the same name will be replaced by the new values.
+ *
+ * E.G
+ *
+ * Input:
+ * |  A |
+ * {B: 1, C: 2}
+ *
+ * Directive
+ * exploder-record :A
+ *
+ * Output:
+ * | A          | A_B | A_C |
+ * {B: 1, C:2}  | 1   |  2 |
+ *
+ */
+@Plugin(type = Directive.TYPE)
+@Name(ExplodeRecord.NAME)
+@Categories(categories = { "column"})
+@Description("Flattens a record into individual columns.")
+public class ExplodeRecord implements Directive, Lineage {
+  public static final String NAME = "explode-record";
+  // Column on which to apply mask.
+  private String[] columns;
+
+  // Map to Keep Track new Columns;
+  private List<String> newColumnNames = new ArrayList<>();
+
+  @Override
+  public UsageDefinition define() {
+    UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
+    builder.define("column", TokenType.COLUMN_NAME_LIST);
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    List<String> cols = ((ColumnNameList) args.value("column")).value();
+    columns = new String[cols.size()];
+    columns = cols.toArray(columns);
+  }
+
+  @Override
+  public void destroy() {
+    // no-op
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context) throws DirectiveExecutionException {
+    List<Row> results = new ArrayList<>();
+
+    for (Row row : rows) {
+      for (String column: columns) {
+        int idx = row.find(column);
+        if (idx == -1) {
+          continue;
+        }
+        Object object = row.getValue(idx);
+
+        if (object == null) {
+          continue;
+        }
+
+        if (!(object instanceof Row)) {
+          throw new DirectiveExecutionException(
+            NAME,
+            String.format("Column '%s' has invalid type '%s'. It should be of type 'Record'.",
+                          column, object.getClass().getSimpleName()));
+        }
+
+        Row rowColumns = (Row) object;
+
+        String columnName;
+        newColumnNames.add((String) column);
+        for (Pair<String, Object> field : rowColumns.getFields()) {
+          columnName = String.format("%s_%s", column, field.getFirst());
+          newColumnNames.add(columnName);
+          row.addOrSet(columnName, field.getSecond());
+        }
+      }
+      results.add(row);
+    }
+    return results;
+  }
+
+  @Override
+  public Mutation lineage() {
```

Comment:
Don't extend this class from `Lineage` since it is impossible to emit lineage correctly for this directive, better not support it rather than giving wrong information